imports/plugins/included/shipping-rates/server/no-meteor/util/locationDenyCheck.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * @summary Filter shipping methods based on per method allow location restrictions
+ * @param {Object} methodRestrictions - method restrictions from FlatRateFulfillmentRestrcitionsCollection
+ * @param {Object} method - current method to check restrcictions against
+ * @param {Object} hydratedOrder - hydrated order for current order
+ * @returns {Bool} true / false as to whether method is still valid after this check
+ */
+export async function locationDenyCheck(methodRestrictions, method, hydratedOrder) {
+  // Get method specific allow restrictions
+  const denyRestrictions = methodRestrictions.filter((methodRestriction) => methodRestriction.type === "deny");
```

Comment:
Same comment as in `attributeDenyCheck` about reducing the number of times we're looping by filtering and then using the filtered array